### ISSUE TITLE
Scan stalls when going into a folder with three dots

### ISSUE THREAD

**Issue by Tak-MK** [tesshucom/jpsonic#2474]
## Problem description

Hi there!
The gist is that everytime jpsonic tries to do the step `[12/22]REFRESH_ALBUM_ID3` / `UPDATE_ORDER_OF_SONG` inside of a folder with three dots, it stalls with this log:
```
java.lang.SecurityException: Access denied to file : /jpsonic/vgm/Shin Megami Tensei/Shin Megami Tensei If.../PS1
        at com.tesshu.jpsonic.service.MediaFileService.getMediaFile(MediaFileService.java:101) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.MediaFileService.getParentOf(MediaFileService.java:139) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.MediaFileService.getParent(MediaFileService.java:143) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.scanner.ScannerProcedureService.albumId3Of(ScannerProcedureService.java:534) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.scanner.ScannerProcedureService.lambda$createAlbumId3s$16(ScannerProcedureService.java:586) ~[classes!/:112.2.11]
        at java.base/java.util.Optional.ifPresent(Unknown Source) ~[na:na]
        at com.tesshu.jpsonic.service.scanner.ScannerProcedureService.createAlbumId3s(ScannerProcedureService.java:585) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.scanner.ScannerProcedureService.refleshAlbumId3(ScannerProcedureService.java:605) ~[classes!/:112.2.11]
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[na:na]
        at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]
        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:798) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:762) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:707) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at com.tesshu.jpsonic.service.scanner.ScannerProcedureService$$EnhancerBySpringCGLIB$$e0310457.refleshAlbumId3(<generated>) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.scanner.MediaScannerServiceImpl.doScanLibrary(MediaScannerServiceImpl.java:173) ~[classes!/:112.2.11]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[na:na]
        at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
---
java.lang.SecurityException: Access denied to file : /jpsonic/music/Bauhaus/Live/This Is for When.../CD1
        at com.tesshu.jpsonic.service.MediaFileService.getMediaFile(MediaFileService.java:101) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.MediaFileService.getParentOf(MediaFileService.java:139) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.MediaFileService.getParent(MediaFileService.java:143) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.scanner.ScannerProcedureService.albumId3Of(ScannerProcedureService.java:534) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.scanner.ScannerProcedureService.lambda$createAlbumId3s$16(ScannerProcedureService.java:586) ~[classes!/:112.2.11]
        at java.base/java.util.Optional.ifPresent(Unknown Source) ~[na:na]
        at com.tesshu.jpsonic.service.scanner.ScannerProcedureService.createAlbumId3s(ScannerProcedureService.java:585) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.scanner.ScannerProcedureService.refleshAlbumId3(ScannerProcedureService.java:605) ~[classes!/:112.2.11]
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[na:na]
        at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]
        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:798) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:762) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:707) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at com.tesshu.jpsonic.service.scanner.ScannerProcedureService$$EnhancerBySpringCGLIB$$d0daf99.refleshAlbumId3(<generated>) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.scanner.MediaScannerServiceImpl.doScanLibrary(MediaScannerServiceImpl.java:173) ~[classes!/:112.2.11]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[na:na]
        at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
---
java.lang.SecurityException: Access denied to file : /jpsonic/music/Porcupine Tree/Arriving Somewhere.../DVD1
        at com.tesshu.jpsonic.service.MediaFileService.getMediaFile(MediaFileService.java:101) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.MediaFileService.getParentOf(MediaFileService.java:139) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.MediaFileService.getParent(MediaFileService.java:143) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.scanner.ScannerProcedureService.albumId3Of(ScannerProcedureService.java:534) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.scanner.ScannerProcedureService.lambda$createAlbumId3s$16(ScannerProcedureService.java:586) ~[classes!/:112.2.11]
        at java.base/java.util.Optional.ifPresent(Unknown Source) ~[na:na]
        at com.tesshu.jpsonic.service.scanner.ScannerProcedureService.createAlbumId3s(ScannerProcedureService.java:585) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.scanner.ScannerProcedureService.refleshAlbumId3(ScannerProcedureService.java:605) ~[classes!/:112.2.11]
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[na:na]
        at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]
        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:798) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:762) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:707) ~[spring-aop-5.3.30.jar!/:5.3.30]
        at com.tesshu.jpsonic.service.scanner.ScannerProcedureService$$EnhancerBySpringCGLIB$$33339b6f.refleshAlbumId3(<generated>) ~[classes!/:112.2.11]
        at com.tesshu.jpsonic.service.scanner.MediaScannerServiceImpl.doScanLibrary(MediaScannerServiceImpl.java:173) ~[classes!/:112.2.11]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[na:na]
        at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
```
Only stalls with one, then I move the folder out, restart jpsonic, and scan again, and it stalls on next folder that has those `...`.
I think all of the folders had mp3 inside instead of flac files, if that helps with anything.

### Steps to reproduce

1. Set up a media folder with any album that has `...` in its name.
2. Scan.
3. See docker logs.

## System information

 * **Jpsonic version**: Jpsonic 112.2.11 – 2023-11-19
 * **Operating system**: Debian stable, using docker jpsonic version
 * **Java version**: 21.0.1
 * **Proxy server**: None
 * **Client**: Chrome 119.0.6045.199（Official Build） （arm64)
 * **Language**: Both English and Japanese
 * **Tomcat/standalone**: None, I'm with docker

**Comment by tesshucom** [comment]
Thank you for your report. We will investigate the cause and release a fixed version soon.

**Comment by tesshucom** [comment]
@Tak-MK

The current policy is to add a process to skip scanning if a file or directory with the ending Dot is encountered. This is a specification that Subsonic and Airsonic do not have, but it is a valid specification. Although it is a rare case, the current problem is that these files are not excluded in Scanning.

Therefore, please do not expect to be able to use your directory "Shin Megami Tensei If..." as is, although it may not meet your needs. This is because the directory name presented is basically a prohibited name in Windows. Of course, we can create such files from commands or programs, but this poses various limitations on the Windows platform. 

> e.g. To use cmd.exe to handle files whose file names end with a period "." on Windows, add the four-character prefix string \\?\ to the beginning of the full path name starting with the drive letter.

Therefore, there are many ripping software that automatically change or remove these prohibited characters. Even if we enable Jpsonic to handle them, there may be various effects when playing them, downloading them, etc. 

#### Workaround

Substitute strings may be available in Unicode. If you really want to reproduce the trailing dot, you'll need a workaround like this.

 - … (Horizontal Ellipsis)
 - ․ (one dot leader)

**Comment by Tak-MK** [comment]
Hi there!

Didn't know about that limitation in Windows with filenames, so I'll add trailing dots to my regex of "change this with a _".

Thanks a lot for the information and your help!

**Comment by tesshucom** [comment]
Some improvements have been made for files ending in Dot. Please note that files ending in Dot will now be skipped from scanning with a warning. This fix is included in v113.0.0. 

 - #2492

Please tell me if you notice something. Thank you.
